hazelcast/serialization/service.py

```diff
@@ -389,6 +389,9 @@ def serializer_for(self, obj):
         # primitives, arrays, String, UUID and some helper types(BigInteger etc)
         serializer = self.lookup_default_serializer(obj_type, obj)
 
+        if serializer is None and obj_type is type:
```

Comment:
We should do it not here but in the broken place. (in `lookup_custom_serializer`)

BTW, the actual problem in `lookup_custom_serializer` is that it considers subclasses whereas it should check the superclasses. If you look at the Java implementation you will see that we consider super types. I would suggest looking there and fixing the actual problem. Then, this will be fixed as well.

Reply:
I see. Done now.